src/plugin_discovery/plugin_exports/reduce_export_specs.js

```diff
@@ -0,0 +1,28 @@
+/**
+ *  Combine the exportSpecs from a list of pluginSpecs
+ *  by calling the reducers for each export type
+ *  @param {Array<PluginSpecs>} pluginSpecs
+ *  @param {Object<exportType,reducer>} exportTypes
+ *  @param {Object<exportType,exports} [defaults={}]
+ *  @return {Object<exportType,exports>}
+ */
+export function reduceExportSpecs(pluginSpecs, reducers, defaults = {}) {
+  return pluginSpecs.reduce((acc, pluginSpec) => {
+    const specsByType = pluginSpec.getExportSpecs() || {};
+    const types = Object.keys(specsByType);
+
+    return types.reduce((acc, type) => {
+      const reducer = (reducers[type] || reducers.unknown);
+
+      if (!reducer) {
+        throw new Error(`Unknown export type ${type}`);
+      }
+
+      const specs = [].concat(specsByType[type] || []);
+
+      return specs.reduce((acc, spec) => (
```

Comment:
🤔

Reply:
I think this is ok.. A lot of times this is how it works when you need to reduce within a reduce within a reduce. Am I just used to that?

Reply:
I'm used to it, though I do wish it wasn't as messy. I think I prefer this to defining the functions outside of the reduce as function definitions or something. I like how you can see the hierarchy here, but if they were all predefined it would be more confusing to see the relationship.

Reply:
:+1: Works for me